src/main/java/org/elasticsearch/script/ScriptModes.java

```
@@ -0,0 +1,215 @@
+                            processedSettings.add(scriptTypePrefix + scriptedOp + ": " + scriptMode);
+                            addScriptMode(scriptEngineService, scriptType, scriptedOp, scriptMode, scriptModes);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static void processDisableDynamicDeprecatedSetting(Settings settings, Map<String, ScriptEngineService> scriptEngines,
+                                                               List<String> processedSettings, Map<String, ScriptMode> scriptModes, ESLogger logger) {
+        //read deprecated disable_dynamic setting, apply only if none of the new settings is used
+        String disableDynamicSetting = settings.get(ScriptService.DISABLE_DYNAMIC_SCRIPTING_SETTING);
+        if (disableDynamicSetting != null) {
+            if (processedSettings.isEmpty()) {
+                ScriptService.DynamicScriptDisabling dynamicScriptDisabling = ScriptService.DynamicScriptDisabling.parse(disableDynamicSetting);
+                switch(dynamicScriptDisabling) {
+                    case EVERYTHING_ALLOWED:
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INDEXED, ScriptMode.ENABLE, scriptModes);
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INLINE, ScriptMode.ENABLE, scriptModes);
+                        break;
+                    case ONLY_DISK_ALLOWED:
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INDEXED, ScriptMode.DISABLE, scriptModes);
+                        addGlobalScriptTypeModes(scriptEngines.keySet(), ScriptType.INLINE, ScriptMode.DISABLE, scriptModes);
+                        break;
+                }
+            } else {
+                logger.warn("ignoring [{}] setting as conflicting scripting settings have been specified: {}", ScriptService.DISABLE_DYNAMIC_SCRIPTING_SETTING, processedSettings);
```

that would be an easy way to get rid of the logger too... :) I discussed this with @clintongormley and we agreed on only logging and ignoring old settings, but sure we can throw an exception and it will make the node startup fail.